cpp/oneapi/dal/algo/pca/test/badarg.cpp

```diff
@@ -0,0 +1,109 @@
+/*******************************************************************************
+* Copyright 2020-2021 Intel Corporation
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*******************************************************************************/
+
+#include <array>
+
+#include "oneapi/dal/algo/pca/infer.hpp"
+#include "oneapi/dal/algo/pca/train.hpp"
+#include "oneapi/dal/table/row_accessor.hpp"
+
+#include "oneapi/dal/test/engine/common.hpp"
+#include "oneapi/dal/test/engine/fixtures.hpp"
+
+namespace oneapi::dal::pca::test {
+
+namespace te = dal::test::engine;
+
+template <typename Method>
+class pca_badarg_test : public te::algo_fixture {
+public:
+    static constexpr std::int64_t row_count = 8;
+    static constexpr std::int64_t column_count = 2;
+    static constexpr std::int64_t element_count = row_count * column_count;
+
+    auto get_descriptor() const {
+        return pca::descriptor<float, Method, pca::task::dim_reduction>{};
+    }
+
+    table get_train_data(std::int64_t override_row_count = row_count,
+                         std::int64_t override_column_count = column_count) const {
+        ONEDAL_ASSERT(override_row_count * override_column_count <= element_count);
```

Comment:
ONEDAL_ASSERT_MUL_OVERFLOW ?

Reply:
This is not ASSERT for overflow

Reply:
I suggest to not check overflows in tests, to me it sounds like overkill. The values in tests are controlled on our side, we can adjust values that do not lead to overflow.